vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/dataview/CheckboxGroupListDataViewImpl.java

```diff
@@ -0,0 +1,152 @@
+package com.vaadin.flow.component.checkbox.dataview;
+
+import com.vaadin.flow.component.checkbox.CheckboxGroup;
+import com.vaadin.flow.data.provider.*;
+import com.vaadin.flow.function.SerializableComparator;
+import com.vaadin.flow.function.SerializableConsumer;
+import com.vaadin.flow.function.SerializableFunction;
+import com.vaadin.flow.function.SerializablePredicate;
+import com.vaadin.flow.shared.Registration;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class CheckboxGroupListDataViewImpl<T> implements CheckboxGroupListDataView<T> {
+
+    private int itemsSize = 0;
+    private Set<SizeChangeListener> sizeChangeListeners;
+    private CheckboxGroup<T> component;
+
+    public CheckboxGroupListDataViewImpl(CheckboxGroup<T> checkboxGroup) {
+        this.component = checkboxGroup;
+    }
+
+    @Override
+    public T getNextItem(T item) {
+        return getNextOrPreviousItem(item, itemsList -> {
+            int itemIndex = itemsList.indexOf(item);
+            if (itemIndex != -1 && itemIndex < itemsList.size() - 1) {
+                return itemsList.get(itemIndex + 1);
+            }
+            return null;
+        });
+    }
+
+    @Override
+    public boolean hasNextItem(T item) {
+        return getNextItem(item) != null;
```

Comment:
Since we always would have a `ListDataProvider` backing things and it provides `Collection<T> getItems()`, this is just an overly complex way of doing `collection.contains(item)`

Reply:
We couldn't use `Collection<T> getItems()` because `hasNextItem()` is supposed to search the item within filtered data as per contract of this method.